service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureVmStep.java

```diff
@@ -109,18 +109,32 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
               .getByResourceGroup(
                   azureCloudContext.getAzureResourceGroupId(), networkResource.getNetworkName());
 
-      var createNic =
-          createNetworkInterface(
-              computeManager,
-              azureCloudContext,
-              existingNetwork,
-              networkResource.getSubnetName(),
-              existingAzureIp);
+      if (!context.getWorkingMap().containsKey(AzureVmHelper.WORKING_MAP_NETWORK_INTERFACE_KEY)) {
```

Comment:
what happens if there're multiple create VM requests at the same time? how does the code know which nic is for which VM?

Reply:
@ddietterich is that the case?

Reply:
From my understanding each flight isolated. So the context is also per flight.

Reply:
Yes, each flight is isolated. Multiple VM creates would be multiple flights.

We have not implemented much concurrency control in WSM. In general, we rely on the metadata transaction to add metadata that will conflict with a parallel operation on the same object. That covers many cases. If we discover a need to cover more cases (like concurrent update and delete operations), then we will need to add some kind of busy state to the metadata in the WSM database.